File: cpp/tests/fixtures/multilang/calculator.cpp
/**
 * Calculator module - C++
 */
#include <vector>
#include <string>

struct Item {
    double price;
    int quantity;
};

struct Result {
    double subtotal;
    double tax;
    double total;
};

Result calculateTotal(const std::vector<Item>& items, double taxRate) {
    // Calculate subtotal
    double subtotal = 0;
    for (const auto& item : items) {
        subtotal += item.price * item.quantity;
    }

    // Apply tax
    double tax = subtotal * taxRate;
    double total = subtotal + tax;

    return Result{subtotal, tax, total};
}

std::string formatCurrency(double amount) {
    char buffer[32];
    snprintf(buffer, sizeof(buffer), "$%.2f", amount);
    return std::string(buffer);
}

bool validateInput(double value) {
    if (value < 0) {
        return false;
    }
    return true;
}

File: cpp/tests/fixtures/multilang/calculator.ts
/**
 * Calculator module - TypeScript
 */

interface Item {
    price: number;
    quantity: number;
}

interface CalculationResult {
    subtotal: number;
    tax: number;
    total: number;
}

function calculateTotal(items: Item[], taxRate: number): CalculationResult {
    // Calculate subtotal
    let subtotal: number = 0;
    for (const item of items) {
        subtotal += item.price * item.quantity;
    }

    // Apply tax
    const tax: number = subtotal * taxRate;
    const total: number = subtotal + tax;

    return {
        subtotal: subtotal,
        tax: tax,
        total: total
    };
}

function formatCurrency(amount: number): string {
    return '$' + amount.toFixed(2);
}

function validateInput(value: unknown): boolean {
    if (value === null || value === undefined) {
        return false;
    }
    if (typeof value !== 'number') {
        return false;
    }
    if (value < 0) {
        return false;
    }
    return true;
}

export { calculateTotal, formatCurrency, validateInput };
